Add debug symbols page for RN (#4425)
getsentry/sentry-docs: src/platforms/react-native/upload-debug.mdx

---
title: Debug Symbols
sidebar_order: 96

> description: "Learn about using ProGuard to upload the ProGuard/R8 mappings, source bundle, and native symbols Sentry."

Reviewer: I may be wrong, but this description doesn't seem to really match what's in the content. 

```suggestion
description: "Learn about different methods for providing debug symbols to Sentry."
```

Author: @imatwawana mm I guess it can be improved, but the suggestion is also not ideal.

There are different type of "debug symbols", there are also different ways to upload it, what's about:

> Learn about using the Sentry Android Gradle Plugin to upload the ProGuard/R8 mappings, source bundle, and debug symbols to Sentry.

Reviewer: The intro to this page includes this sentence: 
 "We offer a range of methods to provide Sentry with debug symbols." My expectation is that you're about to tell me about a range of methods. If you're talking about a range of methods (which it seems like you are because you talk about the out of the box option), then the description of the page should reflect that, no? If this is really just about ProGuard, then we probably need to remove that sentence or follow that with "but on this page we're only going to talk about ProGuard 8" or whatever makes sense.

Author: @imatwawana mm, this sentence was written years ago, don't even remember by whom, but I will try to describe what's supposed to be now.

We have to upload Proguard/R8 mapping file (for obfuscated Java stack traces)
We have to upload debug symbols (for C+/C++ stack traces), the source bundle is an addition to that.

We can do that automatically using the Sentry Android Gradle Plugin.
We can do that via sentry-cli

to be honest I've even suggested renaming this page https://github.com/getsentry/sentry-docs/issues/4395 a few days ago, because its not even about "Proguard" anymore.

Reviewer: Okay, gotcha. Is the point of this to tell users about both methods or just to highlight the plugin?

Author: ideally both, but highlighting the plugin first, since not everyone is able to use the plugin (the majority is tho), that's why there's an alternative, which is thru sentry-cli

Author: I've rewritten it, please have a second look @imatwawana, and sorry for the ping pong.

Reviewer: Looks great!!

---

Debug symbols provide information that Sentry displays on the **Issue Details** page to help you triage an issue. We offer a range of methods to provide Sentry with debug symbols.

## When to Upload

With default settings, complete stack traces are available in your Java/Kotlin error, out of the box, unless you use [enableProguardInReleaseBuilds](https://reactnative.dev/docs/signed-apk-android#enabling-proguard-to-reduce-the-size-of-the-apk-optional). In this case, you need to upload the ProGuard/R8 mapping file generated by the [Android Gradle Plugin](https://developer.android.com/studio/build/shrink-code#shrink-code), so Sentry can show you proper stack traces.

> Errors raised from the native layer in Android apps require certain debug information files to be uploaded. For example, an Android app can use `proguard` for minification and obfuscation. And when using [NDK](https://developer.android.com/ndk/), dwarf debug files need to be uploaded.

Reviewer: ```suggestion
Errors raised from the native layer in Android apps require certain debug information files to be uploaded. For example, an Android app can use `proguard` for minification and obfuscation. Also, when using [NDK](https://developer.android.com/ndk/), dwarf debug files need to be uploaded.
```

> ## Uploading with ProGuard and NDK

Reviewer: ```suggestion
## Uploading With ProGuard and NDK
```

> To use ProGuard or NDK with Sentry, upload the [Android Proguard/R8 mapping files and native symbols](/platforms/android/proguard/) by either the recommended method of using our Gradle integration or manually by using [sentry-cli proguard](/product/cli/dif/#proguard-mapping-upload) or [sentry-cli upload-dif](/product/cli/dif/#uploading-files).

Author: ```suggestion
To use ProGuard or NDK with Sentry, upload the [Android Proguard/R8 mapping files and native symbols](/platforms/android/proguard/) by either the recommended method of using our Gradle integration or manually by using `sentry-cli` [upload-proguard](/product/cli/dif/#proguard-mapping-upload) or [upload-dif](/product/cli/dif/#uploading-files).
```

> To use ProGuard or NDK with Sentry, upload the [Android Proguard/R8 mapping files and native symbols](/platforms/android/proguard/) by either the recommended method of using our Gradle integration or manually by using `sentry-cli` [upload-proguard](/product/cli/dif/#proguard-mapping-upload) or [upload-dif](/product/cli/dif/#uploading-files).

Author: @imatwawana maybe I should split this up into 2 sections, one of them is the Sentry Android Gradle plugin (/platforms/android/proguard/) and the other section with sentry-cli, (/product/cli/dif/) links, so that fits the "range of methods", what do you think?

Reviewer: Depends on the answer to my other question about the main aim of this page.

---
title: Debug Symbols
sidebar_order: 96
description: "Learn about using the Sentry Gradle Plugin or sentry-cli for providing debug symbols to Sentry."
---

Debug symbols provide information that Sentry displays on the **Issue Details** page to help you triage an issue. We offer a range of methods to provide Sentry with debug symbols.

## When to Upload

With default settings, complete stack traces are available in your Java/Kotlin error, out of the box, unless you use [enableProguardInReleaseBuilds](https://reactnative.dev/docs/signed-apk-android#enabling-proguard-to-reduce-the-size-of-the-apk-optional). In this case, you need to upload the ProGuard/R8 mapping files generated by the [Android Gradle Plugin](https://developer.android.com/studio/build/shrink-code#shrink-code), so Sentry can show you proper stack traces.

> Errors raised from the native layer in Android apps require certain debug information files to be uploaded. For example, an Android app with [NDK](https://developer.android.com/ndk/) support, dwarf debug files need to be uploaded.

Reviewer 2: ```suggestion
Errors raised from the native layer in Android apps require certain debug information files to be uploaded. For example, for an Android app with [NDK](https://developer.android.com/ndk/) support, dwarf debug files need to be uploaded.
```
Is here `for` missing or am I not getting the sentence?

Author: I think its ambigual since the sentence starts with `For example`, then, `an Android app...`

Reviewer 2: I've updated the suggestion, maybe it's better like that?

Reviewer: This is one of the few cases where passive voice might be a good idea: 
```suggestion
Errors raised from the native layer in Android apps require certain debug information files to be uploaded. For example, dwarf debug files need to be uploaded for an Android app with [NDK](https://developer.android.com/ndk/) support.
```
